floss/language/go/extract.py

```diff
@@ -0,0 +1,388 @@
+    # Extract string blob in .rdata section that starts with "go:buildid" or "go.buildid"
+    """
+    67 6F 3A 62 75 69 6C 64  69 64 00 69 6E 74 65 72  go:buildid.inter
+    6E 61 6C 2F 63 70 75 2E  49 6E 69 74 69 61 6C 69  nal/cpu.Initiali
+    7A 65 00 69 6E 74 65 72  6E 61 6C 2F 63 70 75 2E  ze.internal/cpu.
+    70 72 6F 63 65 73 73 4F  70 74 69 6F 6E 73 00 69  processOptions.i
+    6E 74 65 72 6E 61 6C 2F  63 70 75 2E 69 6E 64 65  nternal/cpu.inde
+    78 42 79 74 65 00 69 6E  74 65 72 6E 61 6C 2F 63  xByte.internal/c
+    """
+
+    blob_pattern = re.compile(b"go(\.|:)buildid\x00(.)*\x00\x00", re.DOTALL)
+    for m in blob_pattern.finditer(section_data):
+        t = m.group(0)
+        length = 0
+        for s in t.split(b"\x00"):
+            addr = pe.get_offset_from_rva(m.start() + length + section_va)
+            length += len(s) + 1
+            yield from yield_string(s, addr, min_length)
+
+
+def extract_string_blob_in_rdata_data(
+    pe: pefile.PE, section_data, min_length, alignment, fmt
+) -> Iterable[StaticString]:
+    # Extract strings from string table in .rdata section
+    # .data:00537B40                 dd offset unk_4A1E3C
+    # .data:00537B44                 db    4
+    # .data:00537B45                 db    0
+    # .data:00537B46                 db    0
+    # .data:00537B47                 db    0
+    # .data:00537B48                 dd offset unk_4A21C2
+    # .data:00537B4C                 db    6
+    # .data:00537B4D                 db    0
+    # .data:00537B4E                 db    0
+    # .data:00537B4F                 db    0
+
+    for addr in range(0, len(section_data) - alignment // 2, alignment // 2):
+        try:
+            curr = section_data[addr : addr + alignment]
+            s_off, s_size = struct.unpack(fmt, curr)
+
+            if not s_off and not (1 <= s_size < 128):
+                continue
+
+            s_rva = s_off - pe.OPTIONAL_HEADER.ImageBase
+
+            if not pe.get_section_by_rva(s_rva):
+                continue
+
+            binary_string = pe.get_string_at_rva(s_rva, s_size)
+
+            addr = pe.get_offset_from_rva(s_rva)
+
+            yield from yield_string(binary_string, addr, min_length)
+
+        except Exception as e:
+            logger.error(f"Error: {e}")
+            raise
+
+
+def extract_longstrings(
+    pe: pefile.PE, section_data, section_va, min_length, pattern, arch, regex_offset: Optional[int] = 0
+) -> Iterable[StaticString]:
+    for m in pattern.finditer(section_data):
+        s_off = struct.unpack("<I", m.group("offset"))[0]
+        s_size = struct.unpack("<B", m.group("size"))[0]
+
+        if arch == "amd64":
+            s_rva = s_off + m.start() + section_va + regex_offset
+            try:
+                addr = pe.get_offset_from_rva(s_rva)
+            except pefile.PEFormatError:
+                continue
+
+        elif arch == "386":
+            s_rva = s_off - pe.OPTIONAL_HEADER.ImageBase
```

Comment:
I'm confused why the `s_rva` calculations here and above differ. Can you double check and add comments to the code, please?

Reply:
This is better, aren't the actual offsets dependent on the number of opcode bytes though?

Can we get the address/offset via the match object?